cabal2nix/src/Distribution/Nixpkgs/Haskell/FromCabal/License.hs

```diff
@@ -40,128 +41,33 @@ fromCabalLicense AllRightsReserved                         = Known "lib.licenses
 fromCabalLicense (Apache Nothing)                          = Known "lib.licenses.asl20"
 fromCabalLicense (Apache (Just (versionNumbers -> [2,0]))) = Known "lib.licenses.asl20"
 fromCabalLicense ISC                                       = Known "lib.licenses.isc"
-fromCabalLicense OtherLicense                              = Unknown Nothing
+fromCabalLicense OtherLicense                              = Known "lib.licenses.free"
 fromCabalLicense (UnknownLicense "CC0-1.0")                = Known "lib.licenses.cc0"
 fromCabalLicense (UnknownLicense "BSD3ClauseORApache20")   = Known "lib.licenses.bsd3"
 fromCabalLicense l                                         = error $ "Distribution.Nixpkgs.Haskell.FromCabal.License.fromCabalLicense: unknown license"
                                                                   ++ show l ++"\nChoose one of: " ++ intercalate ", " (map display knownLicenses)
 
+fromSPDXLicenseExpression :: SPDX.LicenseExpression -> Distribution.Nixpkgs.License.License
+fromSPDXLicenseExpression  (SPDX.ELicense simpl Nothing) =
+  case simpl of
+  SPDX.ELicenseId lid -> Known ("lib.licensesSpdx.\"" ++ DPretty.prettyShow lid ++ "\"")
+  _ ->
+    -- Not handed: the '+' suffix and user-defined licences references.
+    -- Use the SPDX expression as a free-form license string.
+    Unknown (Just $ DPretty.prettyShow simpl)
+fromSPDXLicenseExpression (SPDX.ELicense simpl (Just excep)) =
+  case simpl of
+  SPDX.ELicenseId lid -> Known ("lib.licensesSpdx.\"" ++ DPretty.prettyShow lid ++ "\" lib.licensesSpdx.\"" ++  DPretty.prettyShow excep ++ "\"")
+  _ ->
+    -- Not handed: the '+' suffix and user-defined licences references.
```

Comment:
```suggestion
    -- Not handled: the '+' suffix and user-defined licences references.
    -- Use the SPDX expression as a free-form license string.
    Unknown (Just $ DPretty.prettyShow simpl)
fromSPDXLicenseExpression (SPDX.ELicense simpl (Just excep)) =
  case simpl of
  SPDX.ELicenseId lid -> Known ("lib.licensesSpdx.\"" ++ DPretty.prettyShow lid ++ "\" lib.licensesSpdx.\"" ++  DPretty.prettyShow excep ++ "\"")
  _ ->
    -- Not handled: the '+' suffix and user-defined licences references.
```

Reply:
There is also a typo in the commit message.